Newsletters: add 82 (2020-01-29)

harding opened the pull request:
- [x] Bitcoin StackExchange section @bitschmidty 
- [x] `C-Lightning #3376` @moneyball

adamjonas commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+  payment.  This differs from current-style LN payments where the
+  timeout and the delay may be chosen independently.
+
+    To allow eltoo to achieve a similar independence of timeout and
```
a similar -> similar

harding replied:
"a similar" feels better to me, but happy to change it if you feel strongly about it.

harding commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+
+- [How is OP_CHECKTEMPLATEVERIFY a scaling solution?]({{bse}}92755)
+  Confused_Coder explains ways in which the proposed new opcode, `OP_CHECKTEMPLATEVERIFY`,
+  could help with more [efficient use of block space over time][news48 output commitments]
```
I think this bullet is a good summary of the answer using the answer's own terms, but I think it's a bit misleading without all of the context from the answer.  The answer describes how CTV works in the case of empty blocks, but in the case of full blocks, CTV is always less efficient than non-CTV to accomplish the same goal.  Maybe s/more efficient use of block space over time/delaying the use of block space until fees are cheaper/.  Also maybe, s/are then later claimed onchain/can later be expanded to multiple outputs/.

bitschmidty replied:
Thanks for the suggestions @harding Ive taken them both as I think they are more correct/clear

jnewbery replied:
I think the new wording is fine, but I think the old wording was fine too. Congestion control transactions would 'help with more efficient use of the block space over time' in the same way that UTXO consolidation helps - by reducing demand when there's competition for block space and moving that demand to a time when there's less competition. If you accept the premise that block space is not uniformly valuable across time, then arbitraging the use of that block space across time is indeed an efficiency win because block space is taken by users who value it more highly.

I'm skeptical that any exchanges would use the proposed congestion control method, and like I said, new wording is also fine :)

jnewbery commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,202 @@
+      script's conditions were fulfilled).  For details, see the BIP or
+      [taproot][topic taproot].
+
+    - [BIP142][] is assigned to "Validation of Taproot Scripts", which
```
Should be BIP342

jnewbery commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+    Although many merges to the BIPs repository include contributions
+    from several different people, this merge had more contributors than
+    any we've seen before: it included content and edits from 30
+    different people in 426 commits and its credits thank several other
```
I only count 163 commits (from merge base commit 24eddbb) and 28 authors (2 duplicates):

```
→ git log 24eddbb.. | grep Author | cut -f2 -d ":" |sort |uniq
 Adam Gibson <ekaggata@gmail.com>
 Agis Anastasopoulos <a@xz0.org>
 andrewtoth <andrewstoth@gmail.com>
 Anthony <pyskell@users.noreply.github.com>
 Anthony Towns <aj@erisian.com.au>
 Bryan Bishop <kanzure@gmail.com>
 codeShark149 <rajarshi149@gmail.com>
 Dev Random <devrandom@users.noreply.github.com>
 Devrandom <miron@basezero.com>
 Dmitry Petukhov <dp@simplexum.com>
 Elichai Turkel <elichai.turkel@gmail.com>
 Fabian Jahr <fabjahr@gmail.com>
 Gregory Sanders <gsanders87@gmail.com>
 Hennadii Stepanov <32963518+hebasto@users.noreply.github.com>
 JamesC <james.chiangwu@gmail.com>
 John Newbery <john@johnnewbery.com>
 Jonas Nick <jonasd.nick@gmail.com>
 Jon Atack <jon@atack.com>
 Kalle Rosenbaum <kalle@rosenbaum.se>
 LaurentMT <LaurentMT@users.noreply.github.com>
 Mark B Lundeberg <36528214+markblundeberg@users.noreply.github.com>
 Matthew Zipkin <pinheadmz@gmail.com>
 Max Hillebrand <30683012+MaxHillebrand@users.noreply.github.com>
 Orfeas Litos <o.thyfronitis@ed.ac.uk>
 Orfeas Stefanos Thyfronitis Litos <o.thyfronitis@ed.ac.uk>
 Pieter Wuille <pieter.wuille@gmail.com>
 stefanwouldgo <stefan@sblbs.de>
 Thomas Kerin <me@thomaskerin.io>
 Tim Ruffing <crypto@timruffing.de>
 Tim Ruffing <tim@timruffing.de>
→ git log 24eddbb.. | grep Author | cut -f2 -d ":" |sort |uniq | wc -l
30
```

harding replied:
Confirmed.  Looks like the major difference was my script was counting merges on sipa's branch.

```text
harding@ganymede:~/repos/bips$ git log --oneline --no-merges  802520e...9cf4038 | wc -l
163
harding@ganymede:~/repos/bips$ git shortlog -s  802520e...9cf4038 | wc -l
30
```

Also, I missed the second duplicate author (I see them now).

Thanks for verifying!

jonatack commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+  linked to, so linking to the PR directly above -->
+
+- [BOLTs #705][] allocates [BOLT1][] message types 32768--65535 for
+  experimental and application specific messages.  It also suggests
```
along with guidelines to follow, including (posting the numbers...)

jonatack commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+- [BOLTs #705][] allocates [BOLT1][] message types 32768--65535 for
+  experimental and application specific messages.  It also suggests
+  anyone allocating themselves a message type from that range post the
+  numbers they're using to [BOLTs #716][] to prevent conflicts.
```
s/BOLTS 716#/BOLTS issue #716/

jonatack commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -85,9 +85,11 @@ endcomment %}
-  could help with more [efficient use of block space over time][news48 output commitments]
-  by encapsulating multiple payments into a single output which are then later claimed onchain.
+  Confused_Coder explains ways in which the proposed new opcode,
+  `OP_CHECKTEMPLATEVERIFY`, could help with more
```
remove "more"

jonatack replied:
or "help [delay..."

bitschmidty commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+      script's conditions were fulfilled).  For details, see the BIP or
+      [taproot][topic taproot].
+
+    - [BIP142][] is assigned to "Validation of Taproot Scripts", which
```
s/142/342

jnewbery commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,217 @@
+  extending the change described in [Newsletter #79][news79 lnd3814].
+  Uneconomical UTXOs are UTXOs that cost more to spend than they contain
+  in value---it's important that LND be able to spend these if the
+  proposed [anchor outputs][news70 anchor] fee bumping method is adopted
```
Can link to /en/topics/anchor-outputs/ now that #318 is merged

harding commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,215 @@
+    - [BIP340][] is assigned to "Schnorr Signatures for secp256k1,"
+      which describes a signature scheme compatible with the secp256k1
+      [elliptic curve][] used by Bitcoin.  The signatures are
+      compatible with batch verification and with key and signature
```
I don't understand.  BIP340 keys and signatures should currently be compatible with batch verification and MuSig.

jonatack replied:
Yes, they are compatible, my thought was to perhaps describe their implementation state.

According to BIP 140, batch signature validation "...code is already in bitcoin/secp256k1 and can be merged in".

That leaves MuSig, which afaik does not have a proposed BIP implementation yet; it is mentioned in the BIPs only in reference to https://eprint.iacr.org/2018/068.

jonatack replied:
Yes, they are compatible, my thought was to perhaps describe their implementation state.

According to BIP 140, batch signature validation "...code is already in bitcoin/secp256k1 and can be merged in".

That leaves MuSig, which afaik does not have a proposed BIP implementation yet; it is mentioned in the BIPs only in reference to https://eprint.iacr.org/2018/068 (and only in BIPs 340-3414-342).

jonatack replied:
and Schnorr key aggregation in https://eprint.iacr.org/2018/068. Happy to be wrong or have misunderstood.

harding replied:
Oh, hmm.  I don't think I want to add details about the implementation state to this section about the BIPs.  I think it's enough to know that the scheme proposed in BIP340 is compatible with MuSig (and perhaps other aggregation schemes).  The scheme chosen by users isn't a consensus detail, so it doesn't need a BIP (though it can have a BIP for standardization).  It's like the difference between using `2 <pubkey> <pubkey> 2 OP_CHECKMULTISG` and `<pubkey> OP_CHECKSIGVERIFY <pubkey> OP_CHECKSIG` --- different scripts with different tradeoffs that accomplish the same goal.

jnewbery replied:
Agree with Dave that this shouldn't be about implementation. This section is about the BIPs, which describe the standard.

jonatack replied:
Thanks -- I appreciate being able to exchange on these things.

harding commented:
Rebased on master so we can link to the *anchor outputs* topic and added a commit that (hopefully) addresses feedback from @jnewbery, @adamjonas, @jonatack, and @bitschmidty (thanks!).

moneyball commented:
Ok I added my commit 726dfe57e1b67c1d17e1f629974c1bf3526d133e.

Many thanks to @harding who both helped me edit the wording and for noob git questions (at least I didn't bother @jnewbery?).

@zmnscpxj in case you'd like to have a look

ZmnSCPxj commented:
@moneyball seems ok to me.

jnewbery commented on _posts/en/newsletters/2020-01-29-newsletter.md:
```diff
@@ -0,0 +1,221 @@
+  which need to assure the ultimate receiver that there's no practical
+  way to revoke or change the output paying them.
+
+- [BIPs #876][] assigns three BIPs, one to each parts of the schnorr-taproot-tapscript
```
s/each parts/each part/

bitschmidty commented:
ACK 500fe1f7fc87a2709525c91bcc236ab5c85c2e11

Looks like the most important comments have been resolved.

bitschmidty commented:
Made a small correction to an absolute link that should have been relative and squashed

harding commented:
ACK fff0b6220d06ea047e3ad0e363f0b21daf0df934 Thanks!
